examples/stochastic_volatility.py

```diff
@@ -85,7 +85,7 @@ def main(args):
     _, fetch = load_dataset(SP500, shuffle=False)
     dates, returns = fetch()
     init_rng_key, sample_rng_key = random.split(random.PRNGKey(args.rng_seed))
-    init_params, potential_fn, constrain_fn = initialize_model(init_rng_key, model, model_args=(returns,))
+    init_params, potential_fn, constrain_fn, _ = initialize_model(init_rng_key, model, model_args=(returns,))
```

Comment:
Haha, I'll revert this back. I think the best option is to return a dict here so we can get more information from the model if needed. But you are right that this is unnecessary change.

Reply:
Hmm, this is used to initialize autoguide too (to save one time we trace the model). Historically, we expose some utilities in `infer` such as `transform_fn`, `constrain_fn`, `find_valid_params`, because they are used at various places and some are used in neutra example. For now, I believe only `initialize_model` is useful and can be used for many inference types: autoguide, kernel (hmc, nuts, sa)... I think it is a good time to decide which is better here in the long term. Do you have a suggestion? (I just want to find a better solution - keeping the old interface is fine to me)

Reply:
I'm fine with changing the return type, but if we are doing that, maybe let us think about what is the minimal, sufficient information that we can return from `initialize_model`, maybe anticipating future requirements. I think a model trace seems like good idea, anything else that we should think about? Do you think we should package this in a `namedtuple` called `models_info` so that it is easier to tack on additional fields in the future?

Reply:
I think that returning a namedtuple is a good solution and can be extended in the future without having to worry about backward compatibility. I don't have an idea right now what fields else that we need. I'll add a TODO note here so that we can revisit this issue when we truly need to extend the return results.

Reply:
If we are using a namedtuple, it doesn't matter as much. It should be easy to add other fields (though I think mostly anything else can be derived from the trace).

Reply:
Just to make sure we are on the same page: I intended to revert the changes and add a TODO here (to return a namedtuple when we need new fields) to preserve backward compatibility for this unnecessary change (I can understand your concern). But I'm happy to make a change to use namedtuple now if you agree (which seems to be so based on your last comment).

Reply:
I think we have sufficient evidence that there is other auxiliary information that will be useful, so I would be in favor of using a namedtuple here (which is a more stable interface anyways). WDYT?

Reply:
Absolutely agreed! Thanks for your suggestions, I'll make the change.